### PR TITLE
Setup github status reporter

### LINKED ISSUE #23: Setup github status reporter

### PR DESCRIPTION
**Pull request by kszucs**
resolves #23

### CODE REVIEW

**Comment by kszucs** [comment]
@ursabot build

**Comment by fsaintjacques** [comment]
Can you status this @ursabot?

**Comment by kszucs** [comment]
@ursabot build

**Comment by kszucs** [comment]
@ursabot build

**Comment by kszucs** [comment]
@ursabot build

**Comment by kszucs** [comment]
@ursabot build

**Comment by kszucs** [comment]
Need to grant permissions for ursabot.

**Comment by kszucs** [comment]
@ursabot build

**Comment by kszucs** [comment]
@ursabot build

**Comment by kszucs** [comment]
@ursabot build

**Comment by kszucs** [comment]
@ursabot build

**Comment by kszucs** [comment]
@ursabot build

**Comment by kszucs** [comment]
@ursabot build

**Comment by kszucs** [comment]
@ursabot build

**Comment by kszucs** [comment]
@ursabot build

**Comment by kszucs** [comment]
@ursabot build

**Comment by fsaintjacques** [comment]
oh shiet, neat!

**Comment by fsaintjacques** [comment]
oops.

**Comment by kszucs** [comment]
@ursabot build

**Comment by kszucs** [comment]
@ursabot build

**Comment by kszucs** [comment]
@fsaintjacques would You like to review then squash&merge if it looks good?

> HINT: both status reports and build trigger via comment works :)

**Comment by kszucs** [comment]
@ursabot build

File: ursabot/hooks.py [hunk 1]
```diff
@@ -35,6 +36,7 @@ def _post(self, url, data):
         client = yield self._client()
         response = yield client.post(url.path, json=data)
         result = yield response.json()
+        log.msg(f'POST to {url} with the following result: {result}')
```
[the remark this answers is not in the document]
Comment (kszucs) [reply ursabot/hooks.py]: `yield response.json()` is like `await response.json()`
the `log` line will be executed only after `result` is ready

File: ursabot/hooks.py [hunk 2]
```diff
@@ -46,48 +48,46 @@ def _parse_command(self, message):
...
+                'pull_request': pull_request,
+                'number': pull_request['number']
+            }, event)
+        except Exception as e:
+            message = "I've failed to start builds for this PR"
+            yield self._post(comments_url, {'body': message})
+            raise e
+        else:
+            message = "I've successfully started builds for this PR"
+            yield self._post(comments_url, {'body': message})
```
[the remark this answers is not in the document]
Comment (kszucs) [reply ursabot/hooks.py]:
```
@inlineCallbacks
def fun():
    something = yield doSomethingAsynchronouslyWith(x)
    return f'{something} is done'
```
The result of `yield fun()` will be '... is done'

File: ursabot/hooks.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kszucs) [reply ursabot/hooks.py]: I'll rewrite these modules to use async/await syntax for better readability.

File: ursabot/hooks.py [hunk 3]
```diff
@@ -46,48 +48,46 @@ def _parse_command(self, message):
 
     @defer.inlineCallbacks
     def handle_issue_comment(self, payload, event):
-        url = payload['issue']['comments_url']
-        repo = payload['repository']
-        body = payload['comment']['body']
-        sender = payload['sender']['login']
-        pull_request = payload['issue'].get('pull_request', None)
-        command = self._parse_command(body)
+        issue = payload['issue']
```
[the remark this answers is not in the document]
Comment (kszucs) [reply ursabot/hooks.py]: Sure, but after We have tests, which is going to be rough to mock/implement...

File: ursabot/hooks.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fsaintjacques) [reply ursabot/hooks.py]: so the function is not async at all?

File: ursabot/hooks.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (fsaintjacques) [reply ursabot/hooks.py]: but here yield is not assigned to a variable.

[the remark this answers is not in the document]
Comment (fsaintjacques) [reply ursabot/hooks.py]:
```
In [1]: def herp(derp):
   ...:     yield derp
   ...:     return "123"
   ...: 

In [2]: herp(456)
Out[2]: <generator object herp at 0x7f2fb006b318>

In [3]: 

In [3]: list(herp(456))
Out[3]: [456]
```

**Pull request merged**